lib/rex/proto/mssql/client.rb

```diff
@@ -71,6 +71,18 @@ def initialize(framework_module, framework, rhost, rport = 1433, proxies = nil)
           @current_database = ''
         end
 
+        # @return [Hash] Detect the platform and architecture of the MSSQL server:
+        #  * :arch [String] The server architecture.
+        #  * :platform [String] The server platform.
+        def detect_platform_and_arch
+          result = {}
+
+          server_vars = query("select @@version")
+          result[:arch] = ARCH_X86_64
+          result[:platform] = server_vars[:rows][0][0].match?('Windows') ? Msf::Platform::Windows.realname : Msf::Platform::Linux.realname
```

Comment:
It'd be good to follow the pattern of the other PRs (postgres+mysql), as well as adding tests :+1: